lib/study.py

```diff
@@ -0,0 +1,248 @@
+import os
+import sys
+import shutil
+import inspect
+import itertools
+import configparser
+import importlib
+import utils
+import mad6t_oneturn
+
+class Study(object):
+
+    def __init__(self, name='study', location='.'):
+        '''Constructor'''
+        self.name = name
+        self.location = location
+        self.config = configparser.ConfigParser()
+        self.config.optionxform = str #preserve case
+        self.mad6t_joblist = []
+        #All the requested parameters for a study
+        self.paths = {}
+        self.madx_params = {}
+        self.madx_input = {}
+        self.madx_output = {}
+        self.oneturn_sixtrack_params = {}
+        self.oneturn_sixtrack_input = {}
+        self.oneturn_sixtrack_output = {}
+        self.sixtrack_params = {}
+        self.sixtrack_input = {}
+        self.sixtrack_output = []
+        self.tables = {}
+        #initialize default values
+        Study.initialize(self)
+
+    def initialize(self):
+        '''initialize a study with some default settings'''
+        self.paths["madx_exe"] = "/afs/cern.ch/user/m/mad/bin/madx"
+        self.paths["sixtrack_exe"] = "/afs/cern.ch/project/sixtrack/build/sixtrack"
+        self.paths["location"] = self.location
+        self.paths["madx_in"] = os.path.join(self.location, "mad6t_input")
+        self.paths["madx_out"] = os.path.join(self.location, "mad6t_output")
+        self.paths["sixtrack_in"] = os.path.join(self.location, "sixtrack_input")
+        self.paths["sixtrack_out"] = os.path.join(self.location, "sixtrack_output")
+        app_path = os.path.abspath(inspect.getfile(Study))
+        app_path = os.path.dirname(os.path.dirname(app_path))
+        tem_path = os.path.join(app_path, 'templates')
+        self.paths["templates"] = tem_path
+
+        os.mkdir(self.paths["madx_in"])
+        os.mkdir(self.paths["madx_out"])
+        os.mkdir(self.paths["sixtrack_in"])
+        os.mkdir(self.paths["sixtrack_out"])
+
+        if os.path.isdir(tem_path):
+             for item in os.listdir(tem_path):
+                 s = os.path.join(tem_path, item)
+                 d = os.path.join(self.location, item)
+                 if os.path.isfile(s):
+                     shutil.copy2(s, d)
+        else:
+            print("Invalid templates folder!")
+        self.madx_output = {
```

Comment:
Honestly, I would decouple the definition of default values from actually structuring the user workspace...

Reply:
Yes, you are right. Maybe we need another function to structure the workspace.  Because the source of templates is also a parameter which can be specified by user.